get-device-spec fails when parent directory is not created with an ambiguous message

**Describe the bug**
Error message can be more descriptive.

It says that the file `store/device-specs/moto-g-5-plus.json` doesn't exists but it actually fails because the `device-specs` directory doesn't exists. When the JSON file isn't there, bundletool creates it OK.

**Bundletool version(s) affected**
Version: 0.8.0

**Stacktrace**
```
[BT:0.8.0] Error: Error while writing the output file 'store/device-specs/moto-g-5-plus.json'.
java.io.UncheckedIOException: Error while writing the output file 'store/device-specs/moto-g-5-plus.json'.
        at com.android.tools.build.bundletool.commands.GetDeviceSpecCommand.writeDeviceSpecToFile(GetDeviceSpecCommand.java:173)
        at com.android.tools.build.bundletool.commands.GetDeviceSpecCommand.execute(GetDeviceSpecCommand.java:161)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:83)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:46)
Caused by: java.nio.file.NoSuchFileException: store/device-specs/moto-g-5-plus.json
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
        at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
        at java.nio.file.Files.newOutputStream(Files.java:216)
        at java.nio.file.Files.write(Files.java:3292)
        at com.android.tools.build.bundletool.commands.GetDeviceSpecCommand.writeDeviceSpecToFile(GetDeviceSpecCommand.java:170)
        ... 3 more
```
**To Reproduce**
Steps to reproduce the behavior.
Include the full command being run as well as, if possible, artifacts the bug can be reproduced with.

> bundletool get-device-spec --output store/device-specs/moto-g-5-plus.json

where:
- directory store already exists
- directory device-specs doesn't exists
- JSON file doesn't exists

**Expected behavior**
It would be awesome to create the whole directories tree. If that's not possible, then it would be great to read something like: "The specified output path is invalid. Is the parent directory of the JSON file already created?"

**Known workaround**
Create the parent directory manually.

**Environment:**
OS: Mac OS 10.13.6

--- a/src/main/java/com/android/tools/build/bundletool/commands/GetDeviceSpecCommand.java
+++ b/src/main/java/com/android/tools/build/bundletool/commands/GetDeviceSpecCommand.java
@@ -168,8 +168,8 @@ public abstract class GetDeviceSpecCommand {
         Files.deleteIfExists(getOutputPath());
       }
 
-      if (Files.notExists(outputFile.getParent())) {
-        Files.createDirectories(outputFile.getParent());
+      if (Files.notExists(getOutputPath().getParent())) {
+        Files.createDirectories(getOutputPath().getParent());
       }
 
       Files.write(outputFile, JsonFormat.printer().print(deviceSpec).getBytes(UTF_8));

--- a/src/main/java/com/android/tools/build/bundletool/commands/GetDeviceSpecCommand.java
+++ b/src/main/java/com/android/tools/build/bundletool/commands/GetDeviceSpecCommand.java
@@ -167,6 +167,11 @@ public abstract class GetDeviceSpecCommand {
       if (getOverwriteOutput()) {
         Files.deleteIfExists(getOutputPath());
       }
+
+      if (Files.notExists(outputFile.getParent())) {
+        Files.createDirectories(outputFile.getParent());
+      }
+
       Files.write(outputFile, JsonFormat.printer().print(deviceSpec).getBytes(UTF_8));
     } catch (IOException e) {
       throw new UncheckedIOException(

--- a/src/test/java/com/android/tools/build/bundletool/commands/GetDeviceSpecCommandTest.java
+++ b/src/test/java/com/android/tools/build/bundletool/commands/GetDeviceSpecCommandTest.java
@@ -264,6 +264,21 @@ public class GetDeviceSpecCommandTest {
   }
 
   @Test
+  public void nonExistentParentDirectory_works() {
+    DeviceSpec deviceSpec = mergeSpecs(sdkVersion(21), density(480), abis("x86"), locales("en-US"));
+
+    Path outputPath = tmp.getRoot().toPath().resolve("non-existent-parent-directory").resolve("device.json");
+    GetDeviceSpecCommand command =
+            GetDeviceSpecCommand.builder()
+                    .setAdbPath(adbPath)
+                    .setAdbServer(fakeServerOneDevice(deviceSpec))
+                    .setOutputPath(outputPath)
+                    .build();
+
+    assertThat(command.execute()).isEqualTo(deviceSpec);
+  }
+
+  @Test
   public void oneDevice_badSdkVersion_throws() throws Exception {
     DeviceSpec deviceSpec =
         mergeSpecs(sdkVersion(1), density(360), abis("x86_64", "x86"), locales("en-US"));

